### PR TITLE
Making thrift, pyhive and tableschema as extra_requires

### PR DESCRIPTION
**Pull request by mistercrunch**
Looking at the dependency tree for license related questions, I noticed
that tableschema had a huge tree, and only people running Hive really
need it. Making this as well as pyhive and thrift optional.

### CODE REVIEW

**Review by john-bodley — APPROVED** [review]
My only comment is whether these optional dependencies should reside in the `requirements.txt` file. I can see the merit in having them there as one could argue it represents the frozen state of the union of all viable configurations.

I speculate we may need to add a `requirements.in` file to ensure that these are capture if someone wanted to regenerate the `requirements.txt` file in the future (see [this](https://github.com/jazzband/pip-tools/issues/331) for context).

**Comment by mistercrunch** [comment]
Good call. I just didn't think about running `pip-compile`. I ran it just now and it removes the deps from `requirements.txt` which I think is what we want.

Now given that the `extra_requires` is optional, we could just be more prescriptive and pin things in that section of `setup.py` if needed/appropriate. Being prescriptive inside something optional is very different than being prescriptive in something that is forced/required.

File: UPDATING.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 This file documents any backwards-incompatible changes in Superset and
 assists people when migrating to a new version.
 
+## Superset 0.32.0
```
Comment (michellethomas) [inline UPDATING.md]: @mistercrunch this PR got into 0.31 so this should be in the 0.31 section. Also I think we should call out the specific dependencies that were removed, we were impacted by removing `flower` would be good to list that out.

Comment (mistercrunch) [reply UPDATING.md]: Oh I assumed this would hit 0.32... let me change that then...

Comment (mistercrunch) [reply UPDATING.md]: Here: https://github.com/apache/incubator-superset/pull/7117

**Pull request merged**